user-guide/Reference/Soft-launch_options/BrokerGateway_Migration.md

```diff
@@ -0,0 +1,158 @@
+
+   At a later stage, the Migration tool might delete the services instead.
+
+1. The *ResetCluster* API (*https://\<ip\>/BrokerGateway/api/clusteringapi/resetcluster*) is called on BrokerGateway to create a new NATS cluster, and the NATS binaries are installed at **C:\Program Files\Skyline Communications\DataMiner BrokerGateway\nats-server** and started as a service called "nats-server".
+
+   The *ResetCluster* API call is only executed by one of the NATSMigration instances, if a cluster is migrated. The node that will execute this command is the one with the lowest IP alphabetically. The other nodes will wait until a NATS session using BrokerGateway can be created.
+
+1. The *MessageBrokerConfig.json* file is written at **C:\ProgramData\Skyline Communications\DataMiner\MessageBrokerConfig.json**.
+
+   This file is used when creating default sessions using the *DataMinerMessageBroker.API(.Native)* library. The file will be rewritten to reference the BrokerGateway URL and API key path. The content of this file can for example look like this:
+
+   ```json
+   {
+     "BrokerGatewayConfig": {
+       "CredentialsUrl": "https://LAURENSVG.skyline.local/BrokerGateway/api/natsconnection/getnatsconnectiondetails",
+       "APIKeyPath": "C:\\Program Files\\Skyline Communications\\DataMiner BrokerGateway\\appsettings.runtime.json"
+     }
+   }
+   ```
+
+> [!NOTE]
+> The NATSMigration tool has a **hard-coded 10-minute timeout** during which the *ResetCluster* operation has to be finished. This is the same time frame you have to start the migration (i.e. enter the "install" command in the tool) on all nodes in the cluster.
+
+> [!IMPORTANT]
+> The NATS configuration (*nats-server.config*) of the NATS instance before the migration is not transferrable to the NATS instance after the migration, to it should **never be copied over**.
+
+## Migrating back to the old system
+
+After you have migrated to BrokerGateway, to go back ot the old system, follow the same procedure as when you run the migration, but use the `uninstall` command instead of the `install` command.
+
+The same restrictions apply as for the migration: this must happen on all DataMiner Agents in the cluster at the same time.
+
+## FAQ
+
+### Can I run a cluster with both SLNet-managed NATS and BrokerGateway-managed NATS?
+
+This is not possible. Both NATS installations use the same network ports, so the services cannot run at the same time on a machine. The credentials these installations use are also different and not compatible with each other, so running SLNet-managed NATS on DMA1 and BrokerGateway-managed NATS on DMA2 will also not function.
+
+### Why should I migrate to BrokerGateway? What are the advantages of this?
+
+Careless NATS communications managed by a decoupled and robust BrokerGateway based on a Single Source of Truth that has the complete knowledge of the cluster. A newer and more performant NATS version is also used and is easier than ever to upgrade.
```

Comment:
Do you mean "Carefree NATS communication"?

Reply:
Yes